Title:

Introduction:

- background information
- state the question
- identify and describe the dataset

Preliminary exploratory data analysis:
- read from the web into R
- Clean and wrangle your data

In [1]:
library(tidyverse)
library(repr)
library(tidymodels)
options(repr.matrix.max.rows = 6)
source('tests.R')
source('cleanup.R')

── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──

✔ ggplot2 3.3.6     ✔ purrr   0.3.4
✔ tibble  3.1.7     ✔ dplyr   1.0.9
✔ tidyr   1.2.0     ✔ stringr 1.4.0
✔ readr   2.1.2     ✔ forcats 0.5.1

── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()

── Attaching packages ────────────────────────────────────── tidymodels 1.0.0 ──

✔ broom        1.0.0     ✔ rsample      1.0.0
✔ dials        1.0.0     ✔ tune         1.0.0
✔ infer        1.0.2     ✔ workflows    1.0.0
✔ modeldata    1.0.0     ✔ workflowsets 1.0.0
✔ parsnip      1.0.0     ✔ yardstick    1.0.0
✔ recipes      1.0.1     

── Conflicts ───────────────────────────────────────── tidymodels_conflicts() ──
✖ scales::discard() masks purrr::discard()
✖ dplyr::filter()   masks stats::filter()
✖ recipes::fixed()  masks stringr::fixed()
✖ dplyr::lag()      masks stats::lag()
✖ yardstick::spec() masks r

ERROR: Error in file(filename, "r", encoding = encoding): cannot open the connection


In [2]:
url_1 <- "https://archive.ics.uci.edu/ml/machine-learning-databases/heart-disease/processed.switzerland.data"
switzerland_data <- read_csv(url_1)
switzerland_data

New names:
• `1` -> `1...2`
• `1` -> `1...3`
• `0` -> `0...5`
• `?` -> `?...6`
• `0` -> `0...7`
• `0` -> `0...9`
• `1` -> `1...11`
• `?` -> `?...12`
• `?` -> `?...13`
• `1` -> `1...14`
Rows: 122 Columns: 14
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (9): 95, ?...6, 0...7, 127, 0...9, .7, 1...11, ?...12, ?...13
dbl (5): 32, 1...2, 1...3, 0...5, 1...14

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


32 1...2 1...3 95  0...5 ?...6 0...7 127 0...9 .7  1...11 ?...12 ?...13
1   34 1     4     115 0     ?     ?     154 0     .2  1      ?      ?     
2   35 1     4     ?   0     ?     0     130 1     ?   ?      ?      7     
3   36 1     4     110 0     ?     0     125 1     1   2      ?      6     
⋮   ⋮  ⋮     ⋮     ⋮   ⋮     ⋮     ⋮     ⋮   ⋮     ⋮   ⋮      ⋮      ⋮     
120 72 1     3     160 0     ?     2     114 0     1.6 2      2      ?     
121 73 0     3     160 0     0     1     121 0     0   1      ?      3     
122 74 1     2     145 0     ?     1     123 0     1.3 1      ?      ?     
    1...14
1   1     
2   3     
3   1     
⋮   ⋮     
120 0     
121 1     
122 1

In [7]:
url_3 <- "https://archive.ics.uci.edu/ml/machine-learning-databases/heart-disease/processed.hungarian.data"
data <- read_delim(url_3, delim = ",")
data

New names:
• `2` -> `2...3`
• `0` -> `0...6`
• `2` -> `2...7`
• `0` -> `0...9`
• `0` -> `0...10`
• `?` -> `?...11`
• `?` -> `?...12`
• `?` -> `?...13`
• `0` -> `0...14`
Rows: 293 Columns: 14
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (9): 130, 132, 0...6, 2...7, 185, 0...9, ?...11, ?...12, ?...13
dbl (5): 28, 1, 2...3, 0...10, 0...14

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


28 1 2...3 130 132 0...6 2...7 185 0...9 0...10 ?...11 ?...12 ?...13 0...14
1   29 1 2     120 243 0     0     160 0     0      ?      ?      ?      0     
2   29 1 2     140 ?   0     0     170 0     0      ?      ?      ?      0     
3   30 0 1     170 237 0     1     170 0     0      ?      ?      6      0     
⋮   ⋮  ⋮ ⋮     ⋮   ⋮   ⋮     ⋮     ⋮   ⋮     ⋮      ⋮      ⋮      ⋮      ⋮     
291 56 1 4     155 342 1     0     150 1     3      2      ?      ?      1     
292 58 0 2     180 393 0     0     110 1     1      2      ?      7      1     
293 65 1 4     130 275 0     1     115 1     1      2      ?      ?      1

- age
- sex
- chest pain type(Value 1: typical angina, Value 2: atypical angina, Value 3: non-anginal pain, Value 4: asymptomatic)
- resting blood pressure
- serum cholestoral in mg/dl
- fasting blood sugar > 120 mg/dl:
    - 1 = true
    - 0 = false
- resting electrocardiographic results: 
    - 0: normal
    - 1: having ST-T wave abnormality (T wave inversions and/or ST elevation or depression of > 0.05 mV)
    - 2: showing probable or definite left ventricular hypertrophy by Estes' criteria
- maximum heart rate achieved
- exercise induced angina:
    - 1 = yes
    - 0 = no
- ST depression induced by exercise relative to rest
- the slope of the peak exercise ST segment: 
    - 1: upsloping
    - 2: flat
    - 3: downsloping
- number of major vessels (0-3) colored by flourosopy
- 3 = normal; 6 = fixed defect; 7 = reversable defect
- diagnosis of heart disease (angiographic disease status)：
    - 0: < 50% diameter narrowing
    - 1: > 50% diameter narrowing

Method:

Expected outcomes and significance:
- expect to find
- impact
- future questions